### DISCUSSION TITLE
Span Error when Saving Large Model

### DISCUSSION THREAD

**Discussion by pwylcrow** [xamarin/TorchSharp#1219] [General]
I'm getting an error when trying to save a very large model:

Unhandled exception. System.ArgumentException: Span only supports up to int.MaxValue elements. at TorchSharp.torch.Tensor._validate(Int64 totalSize)

**Comment by NiklasGustafsson** [comment]
Ouch! Maybe @shaltielshmid has some thoughts?

**Reply by shaltielshmid** [reply]
Interesting. 
We can create a function "bytesToStream" and "bytesFromStream", and then when loading copy over using copy_ (this method works across devices, so we can hit two birds with one stone here).

**Reply by shaltielshmid** [reply]
@NiklasGustafsson What do you think? 
I'll try to create a POC in a PR (and then add that into [TorchSharp.PyBridge](https://github.com/shaltielshmid/TorchSharp.PyBridge) as well, so that it would work with the other formats as well).

**Reply by shaltielshmid** [reply]
Added PR here: https://github.com/dotnet/TorchSharp/pull/1234
